[WPE] Build failure with Clang 4.0.1: no matching conversion for functional-style cast from 'pointer' (aka 'unsigned short *') to 'WTF::String'
https://bugs.webkit.org/show_bug.cgi?id=175477
Reviewed by NOBODY (OOPS!).

* Shared/wpe/WebEventFactory.cpp:
(WebKit::singleCharacterStringForKeyEvent): Cast to UChar* to make
compiler choose the correct WTF::String constructor.

--- a/Source/WebKit2/Shared/wpe/WebEventFactory.cpp	
+++ a/Source/WebKit2/Shared/wpe/WebEventFactory.cpp	
@@ -59,7 +59,7 @@ static String singleCharacterStringForKeyEvent(struct wpe_input_keyboard_event*
     glong length;
     GUniquePtr<gunichar2> uchar16(g_ucs4_to_utf16(&event->unicode, 1, 0, &length, nullptr));
     if (uchar16)
-        return String(uchar16.get());
+        return String(reinterpret_cast<UChar*>(uchar16.get()), length);
     return String();
 }
 
